tempto-core/src/main/java/io/trino/tempto/assertions/QueryAssert.java

```diff
@@ -89,6 +94,19 @@ public static QueryExecutionAssert assertThat(QueryCallback queryCallback)
         return new QueryExecutionAssert(ofNullable(executionException));
     }
 
+    @CheckReturnValue
+    public static AbstractThrowableAssert<?, ? extends Throwable> assertQueryFailure(QueryCallback queryCallback)
+    {
+        QueryExecutionException executionException = null;
```

Comment:
```
return assertThatThrownBy(() -> queryCallback.executeQuery())
       .isInstanceOf(QueryExecutionException.class)
```

Reply:
No, this wouldn't be the same. The point here is that i unwrap `QueryExecutionException` cause.

Reply:
Yeah. I did not notice the unwrapping. Make sense then